editor/src/templates/editor.tsx

```diff
@@ -357,47 +373,65 @@ export class Editor {
           }
 
           const urlParams = new URLSearchParams(window.location.search)
-          const githubOwner = urlParams.get('github_owner')
-          const githubRepo = urlParams.get('github_repo')
+          const githubBranch = urlParams.get('github_branch')
           const importURL = urlParams.get('import_url')
 
+          const githubRepoObj: GithubRepo | null = (() => {
```

Comment:
Could this just be turned into a regular function and have it be invoked here? Feels like a lot of URL handling logic all embedded within the `Editor` component.

Reply:
yeah very good point, it's became too massive